torch/nn/parallel/distributed.py

```diff
@@ -188,6 +197,34 @@ def train(self, mode=True):
         for module in self._module_copies[1:]:
             module.train(mode)
 
+    def _dist_broadcast_coalesced(self, tensors, buffer_size):
+        """
+        Broadcast a sequence of tensors to the default group from rank 0.
+        Small tensors are first coalesced into a buffer to reduce the number of
+        broadcasts.
+
+        tensors (sequence): tensors to broadcast. Each tensor needs to be on the
+                            same GPU.
+        buffer_size (int): maximum size of the buffer for coalescing
+        """
+        tensors_bucket = []
+        # To init the first bucket right away
+        cur_bucket_size = buffer_size
+        for tensor in tensors:
+            if cur_bucket_size >= buffer_size:
+                tensors_bucket.append([])
+                cur_bucket_size = 0
+            tensors_bucket[-1].append(tensor)
+            cur_bucket_size += tensor.numel() * tensor.element_size()
```

Comment:
There's `torch._utils._take_tensors` that will do most of these things for you, including handling different types, which is not handled here. Please use that. To see an example look at how `broadcast_coalesced` [used to be implemented](https://github.com/pytorch/pytorch/blob/fa5efab669d68f2931292176d3376e3519bdd0b4/torch/cuda/comm.py#L32-L70).